packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -1559,25 +1668,50 @@ class _NavigationBarTransition extends StatelessWidget {
     @required this.animation,
     @required this.topNavBar,
     @required this.bottomNavBar,
-  }) : heightTween = Tween<double>(
-         begin: bottomNavBar.renderBox.size.height,
-         end: topNavBar.renderBox.size.height,
-       ),
-       backgroundTween = ColorTween(
-         begin: bottomNavBar.backgroundColor,
-         end: topNavBar.backgroundColor,
-       ),
-       borderTween = BorderTween(
-         begin: bottomNavBar.border,
-         end: topNavBar.border,
-       );
+  })
+      : heightTween = Tween<double>(
+    begin: bottomNavBar.renderBox.size.height,
+    end: topNavBar.renderBox.size.height,
+  ),
+        backgroundTween = LinearGradientTween(
```

Comment:
rather than cherry-picking contents of the box decoration, you should just tween between the box decoration instance altogether. You can see that BoxDecoration.lerp already lerps Gradient.lerp. Generic logic is preferred.